src/lib/pages.ts

```diff
@@ -141,7 +141,15 @@ function resolvePageDocument(
     if (page.type === 'group') {
         const firstDocument = resolveFirstDocument(page.pages, [...ancestors, page]);
         if (firstDocument) {
-            return firstDocument;
+            const { page: firstPageInGroup, ancestors } = firstDocument;
+            return {
+                // Keep the page group title but use the first page in the group for the rest.
+                page: {
+                    ...firstPageInGroup,
+                    title: page.title,
+                },
+                ancestors,
+            };
```

Comment:
It seems a bit risky to transform data on the fly in a low function like this one that is not used only for resolving content refs.

Do we have alternatives?

Reply:
Good point. I've changed it to handle it in resolveContentRef:
https://github.com/GitbookIO/gitbook/pull/2381/commits/ce569ca218165f8a16a8ae98602994e4a3325f48

Let me know what you think.